resources/sdk/clisdkclient/extensions/services/commandservice.go

```diff
@@ -0,0 +1,180 @@
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+	"gc/config"
+	"gc/models"
+	"gc/restclient"
+	"gc/utils"
+	"log"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+//CommandService holds the method signatures for all common Command object invocations
+type CommandService interface {
+	Get(uri string) (string, error)
+	List(uri string) (string, error)
+	Post(uri string, payload string) (string, error)
+	Patch(uri string, payload string) (string, error)
+	Put(uri string, payload string) (string, error)
+	Delete(uri string) (string, error)
+	DetermineAction(httpMethod string, operationId string, uri string) (string, error)
+}
+
+type commandService struct {
+	cmd *cobra.Command
+}
+
+//NewCommandService initializes a new command Service object
+func NewCommandService(cmd *cobra.Command) *commandService {
+	return &commandService{cmd: cmd}
+}
+
+func (c *commandService) Get(uri string) (string, error) {
+	profileName, _ := c.cmd.Root().Flags().GetString("profile")
+	config, err := config.GetConfig(profileName)
+	if err != nil {
+		return "", err
+	}
+
+	restClient := restclient.NewRESTClient(config)
+
+	response, err := restClient.Get(uri)
+
+	if err != nil {
+		return "", err
+	}
+
+	return response, nil
+}
+
+func (c *commandService) List(uri string) (string, error) {
+	profileName, _ := c.cmd.Root().Flags().GetString("profile")
+	config, err := config.GetConfig(profileName)
+	if err != nil {
+		return "", err
+	}
+
+	restClient := restclient.NewRESTClient(config)
+
+	//Looks up first page
+	data, err := restClient.Get(uri)
+	if err != nil {
+		return "", err
+	}
+
+	firstPage := &models.Entities{}
+	json.Unmarshal([]byte(data), firstPage)
+
+	//Allocate the total results based on the page count
+	totalResults := make([]string, len(firstPage.Entities))
+
+	//Appends the individual records from Entities into the array
+	for _, val := range firstPage.Entities {
+		totalResults = append(totalResults, string(val))
+	}
+
+    //Looks up the rest of the pages
+	if firstPage.PageCount > 1 {
+		pagedURI := uri
+		for x := 2; x <= firstPage.PageCount; x++ {
+			if strings.Contains(pagedURI,"pageNumber=") {
```

Comment:
@carnellj Some changes here for the query string because sometimes the pageNumber will already be part of the URI passed into this method.

Reply:
Sounds good